### PR TITLE
Create executable and do version checking

### LINKED ISSUE #6: Create executable with versions and releases
I want to enable users to download and run an executable file in an effort to make using the bot as easy as possible, even to non-technical users.

For that we need
- A way to compile an executable (pyinstaller for Windows)
- A version file
- A check for updates and a notifier for updates
- Releases
- Have a 'Press any key to exit' catch for the executable, so people are able to catch errors in the console output

### PR DESCRIPTION
**Pull request by akshualy**
Resolves #6 

Code changes:
- Adds `pyinstaller` to create a windows binary, including an icon file
- Adds file resolving, since pyinstaller creates files in a different location
- Adds a function to compare two semantic versions
  - Used to compare the Play Store with the emulator TFT version
  - Used to compare remote bot version with the local one
- Renames the optional dependency group 'lint' to 'dev' - Install now needs to be done with `pip install .[dev]`
- Creates a logger sink to a new `logs` folder, up to 10 most recent logs are being kept there

Since we now keep logs, I decided against a "Press any key to continue" catch in the console. It was not as reliable as I wanted it to be.

Workflow changes:
- Adds a 'build' workflow to build the executable
  - Comes with a release workflow which needs to be triggered manually on main
  - The release workflow grabs the version from the `pyproject.toml` version field
- Renamed the 'main' workflow to 'lint' and now runs on ubuntu

README changes:
- Now distinguishes between executable and source for setup and running
- Adds a section about how to compile
- Changes the 'Lint' badge to the 'Build' badge

### CODE REVIEW

File: Alune.spec [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# -*- mode: python ; coding: utf-8 -*-
+
+
+a = Analysis(
+    ['main.py'],
+    pathex=[],
+    binaries=[],
+    datas=[('alune/images', 'alune/images')],
```
Comment (Kyrluckechuck) [inline Alune.spec:8]: Is this duplication necessary?

File: Alune.spec [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    hiddenimports=[],
+    hookspath=[],
+    hooksconfig={},
+    runtime_hooks=[],
+    excludes=[],
+    noarchive=False,
+    optimize=0,
```
Comment (Kyrluckechuck) [inline Alune.spec:15]: Any reason not to further optimize the code and reduce the file size?
https://pyinstaller.org/en/stable/feature-notes.html?highlight=bytecode%20optimization#bytecode-optimization-level

**Review by Kyrluckechuck — APPROVED** [review]
Amazing work! Happy to see all that boilerplate and debugging done on the previous bot can be carried on here, though you added some much cleaner improvements 👌

**Review by Detergent13 — APPROVED** [review]
LGTM for the bot changes, although I'm probably not qualified to comment on the CI portion.

File: Alune.spec (hunk 1 above)
Comment (akshualy) [reply Alune.spec:8]: Yes, at least as far as I understood it, it's (origin, target).

**Comment by akshualy** [comment]
> Happy to see all that boilerplate and debugging done on the previous bot can be carried on here
Honestly a life saver, could probably not have bothered without already knowing the pit falls

> although I'm probably not qualified to comment on the CI portion.
No worries, I also copied most of it from the previous CI flows in the fork (mostly done by Kyrluckechuck). If you want to talk/learn about it, you know which group to drop a message in :)

File: Alune.spec (hunk 2 above)
Comment (Kyrluckechuck) [reply Alune.spec:15]: For now if that's all, let's just leave it yeah! 😛

**Pull request merged**